src/amo/reducers/addonsByAuthors.js

```diff
@@ -228,6 +228,33 @@ const reducer = (
           ...newState.byAddonSlug,
           [forAddonSlug]: undefined,
         };
+      } else if (authorUsernames.length) {
```

Comment:
It will be surprising that this reducer behaves differently in the presence of `forAddonSlug`. Is it not possible to do it like this?

```diff
diff --git a/src/amo/reducers/addonsByAuthors.js b/src/amo/reducers/addonsByAuthors.js
index bc184e7db..2cd024516 100644
--- a/src/amo/reducers/addonsByAuthors.js
+++ b/src/amo/reducers/addonsByAuthors.js
@@ -228,7 +228,9 @@ const reducer = (
           ...newState.byAddonSlug,
           [forAddonSlug]: undefined,
         };
-      } else if (authorUsernames.length) {
+      }
+
+      if (authorUsernames.length) {
         // Potentially remove add-ons loaded for these authors with this add-on
         // type, so that we can load new add-ons in the UI (pagination). We do
         // not do it when `forAddonSlug` is defined to avoid re-fetching the
```

I know a test fails but I'm looking at `Addon` and I'm not sure if it matters or not. If it *does* matter then that tells me we need a new action instead of re-using the `FetchAddonsByAuthor` action. 

However, I don't think it's worth holding up your patch for such a big change. It would be fine to land it as is. 

I just wanted to point out that this could potentially create confusion in the future if someone were not intimately familiar with the implementation of this action/reducer logic.

Reply:
No, I don't think it matters. The only reason I've done that is because I did not want to reset the state for the "add-ons by authors" on the add-on detail page to avoid re-fetching those add-ons. Apart from that, there is no reason not to do it.

Reply:
> The only reason I've done that is because I did not want to reset the state for the "add-ons by authors" on the add-on detail page to avoid re-fetching those add-ons.

That sounds like a premature optimization and one that might be better fixed a different way (if it ever becomes a problem).

Reply:
Right, I don't really see this as a premature optimization because it was the old behavior but I agree that we can fix it in a different way if needed.